store.go

```diff
@@ -240,3 +265,79 @@ func (s *SafeConfigStore) Size() int {
 	defer s.mu.Unlock()
 	return s.c.Size()
 }
+
+// LogableConfigStore implements the ConfigStore interface and provides a store
+// that is safe to read and write from multiple goroutines.
+type LogableConfigStore struct {
+	c  *DefaultConfigStore
+}
+
+// NewLogableConfigStore takes a LoggingInterface and returns a new ConfigStore
+// with said interface as the logging mechanism used for read and writes.
+func NewLogableConfigStoreWith(l LoggingInterface) ConfigStore {
+	return &LogableConfigStore{
+		c: newLogableConfigStoreWith(l),
+	}
+}
+
+// NewLogableConfigStore returns a ConfigStore with a default logging mechanism
+// set to write to os.Stdout.
+func NewLogableConfigStore() ConfigStore {
+	l := log.New(os.Stdout, "", 0)
+	return &LogableConfigStore{
+		c: newLogableConfigStoreWith(l),
+	}
+}
+
+// RegisterResolver registers a custom config resolver for the specified
+// ConfigLevel.
+//
+// Additionally, if the provided level is not already in the current collection
+// of active config levels, it will be added automatically
+func (l *LogableConfigStore) RegisterResolver(level ConfigLevel, r Resolver) {
+	l.c.RegisterResolver(level, r)
+}
+
+// SetLevel is a generic key/value setter method. It sets the provided k/v at
+// the specified level inside the map, conditionally creating a new ConfigMap if
+// one didn't previously exist.
+func (l *LogableConfigStore) SetLevel(level ConfigLevel, key string, value interface{}) {
+	l.c.SetLevel(level, key, value)
```

Comment:
So if you move the logging bits from the `DefaultConfigStore` here, then you can get the same behavior but without 1) incurring the overhead of logging in the `DefaultConfigStore` and 2) have the ability to dynamically apply the debug logger onto any `ConfigStore` (once the type of `l.c` is changed)